src/main/java/uk/gov/beis/els/categories/waterheaters/model/ConventionalWaterHeatersForm.java

```diff
@@ -1,56 +1,75 @@
 import javax.validation.constraints.NotNull;
 import javax.validation.groups.Default;
 import org.hibernate.validator.group.GroupSequenceProvider;
+import uk.gov.beis.els.api.common.ApiValuesFromEnum;
+import uk.gov.beis.els.api.common.ApiValuesFromLegislationCategory;
+import uk.gov.beis.els.categories.common.LoadProfile;
 import uk.gov.beis.els.categories.common.StandardTemplateForm30Char;
 import uk.gov.beis.els.categories.internetlabelling.model.InternetLabellingGroup;
 import uk.gov.beis.els.categories.waterheaters.model.validation.ConsumptionUnitBoth;
 import uk.gov.beis.els.categories.waterheaters.model.validation.ConsumptionUnitGj;
 import uk.gov.beis.els.categories.waterheaters.model.validation.ConsumptionUnitKw;
 import uk.gov.beis.els.categories.waterheaters.model.validation.ConventionalWaterHeaterFormSequenceProvider;
+import uk.gov.beis.els.categories.waterheaters.service.WaterHeatersService;
 import uk.gov.beis.els.model.meta.DualModeField;
 import uk.gov.beis.els.model.meta.FieldPrompt;
 import uk.gov.beis.els.model.meta.StaticProductText;
 
+@Schema(name = "Conventional water heater energy label")
 @StaticProductText("You must display the label at the point of sale so that it’s easy to see and clearly related to the product. It must be at least 105mm x 200mm when printed.")
 @GroupSequenceProvider(ConventionalWaterHeaterFormSequenceProvider.class)
 public class ConventionalWaterHeatersForm extends StandardTemplateForm30Char {
 
   @FieldPrompt("Declared load profile")
   @NotBlank(message = "Select a declared load profile")
+  @ApiValuesFromEnum(value = LoadProfile.class)
   private String declaredLoadProfile;
 
   @FieldPrompt("Water heating energy efficiency class")
   @NotBlank(message = "Select an energy efficiency indicator", groups = {Default.class, InternetLabellingGroup.class})
   @DualModeField
+  @ApiValuesFromLegislationCategory(serviceClass = WaterHeatersService.class)
   private String efficiencyRating;
 
   @FieldPrompt("How do you want to provide annual energy consumption?")
   @NotBlank(message = "Specify how you want to provide annual energy consumption")
+  @ApiValuesFromEnum(EnergyConsumptionUnit.class)
   private String consumptionUnit;
 
   @FieldPrompt("Annual electricity consumption in kWh/annum")
   @Digits(integer = 4, fraction = 0, message = "Enter the annual electricity consumption, up to 4 digits long", groups = ConsumptionUnitKw.class)
+  @NotNull(groups = ConsumptionUnitKw.class)
+  @Schema(type = "integer")
   private String kwhAnnum;
 
   @FieldPrompt("Annual fuel consumption in GJ/annum")
   @Digits(integer = 2, fraction = 0, message = "Enter the annual fuel consumption, up to 2 digits long", groups = ConsumptionUnitGj.class)
+  @NotNull(groups = ConsumptionUnitGj.class)
+  @Schema(type = "integer")
   private String gjAnnum;
 
   // These need to be separate otherwise the kw/gj fields above are treated as lists because they appear multiple times in the dom
```

Comment:
I wonder if its worth creating a sideways copy of this form for the API so we can clean up these fields a bit.
Currently you get all these fields in the API doc:
```  
"consumptionUnit": "KWH",
"kwhAnnum": 2,
"gjAnnum": 3,
"bothKwhAnnum": 0,
"bothGjAnnum": 0
```
and so if you want to provide KWH, GJ or both you have to set different fields which feels a bit messy:
```
// kwh only
"consumptionUnit": "KWH",
"kwhAnnum": 2

// gj only
"consumptionUnit": "GJ",
"gjAnnum": 3,

// both
"consumptionUnit": "BOTH",
"bothKwhAnnum": 0,
"bothGjAnnum": 0
```

I think ideally we'd just have 2 field, `kwhAnnum` and `gjAnnum` and they set either/both. Any thoughts @matteason